### PR TITLE
feat(unlock-app)- query for all keys (expired + active)

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
Add query foll all keys (expired + active)  

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/queries/keyholdersByLock.ts [hunk 1]
```diff
@@ -58,6 +87,20 @@ const ExpiredKeys = gql`
   }
 `
 
-export default function keyholdersByLockQuery(showActive = true) {
-  return showActive ? ActiveKeys : ExpiredKeys
+const QUERY_BY_TYPE = {
+  [MemberFilters.ACTIVE]: ActiveKeys,
+  [MemberFilters.EXPIRED]: ExpiredKeys,
+  [MemberFilters.ALL]: AllKeys,
+}
```
Comment (searchableguy) [inline unlock-app/src/queries/keyholdersByLock.ts:94]: Not a big deal for this but we should deprecate use of enum (they are not recommended to be used by the typescript team)

File: unlock-app/src/services/graphService.ts [hunk 2]
```diff
@@ -62,11 +62,10 @@ export class GraphService {
     search: string | number
     filterKey: string
   }) => {
-    const showActive = expiration === MemberFilters.ACTIVE
     const query =
       filterKey === 'keyId'
         ? keyholdersByKeyIdQuery()
-        : keyHoldersByLocks(showActive)
+        : keyHoldersByLocks(expiration)
```
Comment (searchableguy) [inline unlock-app/src/services/graphService.ts:68]: Some explanation would be helpful.

File: unlock-app/src/queries/keyholdersByLock.ts (hunk 1 above)
Comment (kalidiagne) [reply unlock-app/src/queries/keyholdersByLock.ts]: remove the enum with `type` definition

File: unlock-app/src/queries/keyholdersByLock.ts [hunk 3]
```diff
@@ -1,4 +1,33 @@
-import { gql } from 'apollo-boost'
+import { DocumentNode, gql } from 'apollo-boost'
+import { MemberFilter } from '~/unlockTypes'
+
+const AllKeys = gql`
```
Comment (julien51) [inline unlock-app/src/queries/keyholdersByLock.ts:4]: Let's move to another file to keep the name consistent

**Review by julien51 — COMMENTED** [review]
Please check tests

**Review by julien51 — APPROVED** [review]

File: unlock-app/src/components/content/MembersContent.tsx [hunk 4]
```diff
@@ -140,7 +140,7 @@ const filters: Filter[] = [
   {
     key: 'expiration',
     label: 'Expiration',
-    options: [MemberFilters.ACTIVE, MemberFilters.EXPIRED, MemberFilters.ALL],
+    options: ['active', 'expired', 'all'],
```
Comment (searchableguy) [inline unlock-app/src/components/content/MembersContent.tsx:143]: you can actually type this btw. 

```ts
type Filters = "Active" | "Expired" | "All"

// use it elsewhere

options: Filters[]
option: Filter

```

Comment (kalidiagne) [reply unlock-app/src/components/content/MembersContent.tsx:143]: oh, i do it by purpose, but eventually can be changed when we need different filters!

**Pull request merged**